### ISSUE TITLE
How is the confidence modelled? 

### ISSUE THREAD

**Issue by Zador-Pataki** [YvanYin/Metric3D#45]
Hi, thanks for the great work. I am curious about how you have trained your network to [predict confidence](https://github.com/YvanYin/Metric3D/blob/377e6c6642d0aca7aaa5a19e58fdcf5d0fd3d910/mono/model/decode_heads/HourGlassDecoder.py#L253C4-L253C4). Does one of the losses mentioned in the paper supervise the confidence estimate? If not, then could you please provide the loss function you used to supervise the confidence? 

Furthermore, it seems like the confidence is appropriately predicted in the range [0, 1], however, [the output of the network block that outputs the confidences is only a convolutional layer](https://github.com/YvanYin/Metric3D/blob/377e6c6642d0aca7aaa5a19e58fdcf5d0fd3d910/mono/model/decode_heads/HourGlassDecoder.py#L144C9-L144C25), and I see no sigmoid layer. How come? 

Thanks!

**Comment by JUGGHM** [comment]
> Hi, thanks for the great work. I am curious about how you have trained your network to [predict confidence](https://github.com/YvanYin/Metric3D/blob/377e6c6642d0aca7aaa5a19e58fdcf5d0fd3d910/mono/model/decode_heads/HourGlassDecoder.py#L253C4-L253C4). Does one of the losses mentioned in the paper supervise the confidence estimate? If not, then could you please provide the loss function you used to supervise the confidence?
> 
> Furthermore, it seems like the confidence is appropriately predicted in the range [0, 1], however, [the output of the network block that outputs the confidences is only a convolutional layer](https://github.com/YvanYin/Metric3D/blob/377e6c6642d0aca7aaa5a19e58fdcf5d0fd3d910/mono/model/decode_heads/HourGlassDecoder.py#L144C9-L144C25), and I see no sigmoid layer. How come?
> 
> Thanks!

Thanks for your interest! It was supervised by:  `1-|depth_gt-depth_pred|/depth_gt` with L1 loss.
It should have been constrained into 0-1 by Sigmoid function, while is not the case due to code legacy issues.
